x-pack/solutions/security/plugins/security_solution/public/timelines/pages/timelines_page.tsx

```diff
@@ -16,15 +16,29 @@ import { useUserPrivileges } from '../../common/components/user_privileges';
 import { StatefulOpenTimeline } from '../components/open_timeline';
 import * as i18n from './translations';
 import { SecurityPageName } from '../../app/types';
-import { useSourcererDataView } from '../../sourcerer/containers';
 import { EmptyPrompt } from '../../common/components/empty_prompt';
 import { SecurityRoutePageWrapper } from '../../common/components/security_route_page_wrapper';
+import { DataViewManagerScopeName } from '../../data_view_manager/constants';
+import { useSourcererDataView } from '../../sourcerer/containers';
+import { useIsExperimentalFeatureEnabled } from '../../common/hooks/use_experimental_features';
+import { useFullDataView } from '../../data_view_manager/hooks/use_full_data_view';
 
 export const DEFAULT_SEARCH_RESULTS_PER_PAGE = 10;
 
 export const TimelinesPage = React.memo(() => {
   const { tabName } = useParams<{ pageName: SecurityPageName; tabName: string }>();
-  const { indicesExist } = useSourcererDataView();
+
+  const newDataViewPickerEnabled = useIsExperimentalFeatureEnabled('newDataViewPickerEnabled');
+  let { indicesExist } = useSourcererDataView();
+
+  const { dataView } = useFullDataView(DataViewManagerScopeName.default);
+  // NOTE: there should be a Suspense / some kind of loader here as this value is not settled immediately
```

Comment:
Is this what you'd want to do with an HOC for all of our primary pages?

Reply:
yeah, I think so:)